scripts/core/editor3/components/Editor3.jsx

```diff
@@ -0,0 +1,224 @@
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {Editor, EditorState, RichUtils, ContentState} from 'draft-js';
+
+export class Editor3 extends React.Component {
+  constructor(props) {
+    super(props);
+    var initialContentState = ContentState.createFromText(props.value);
+    this.state = {editorState: EditorState.createWithContent(initialContentState)};
+
+    this.readOnly = props.readOnly || false;
+    this.showToolbar = props.showToolbar || false;
+    this.editorFormat = props.editorFormat || [];
+    this.parentOnChange = props.onChange;
+
+    this.focus = () => this.refs.editor.focus();
+    this.onChange = (editorState) => this._onChange(editorState);
+
+    this.handleKeyCommand = (command) => this._handleKeyCommand(command);
+    this.onTab = (e) => this._onTab(e);
+    this.toggleBlockType = (type) => this._toggleBlockType(type);
+    this.toggleInlineStyle = (style) => this._toggleInlineStyle(style);
+  }
+
+  _onChange(editorState) {
+    this.setState({editorState});
+    this.parentOnChange(editorState.getCurrentContent().getPlainText());
+  }
+
+  _handleKeyCommand(command) {
+    const {editorState} = this.state;
+    const newState = RichUtils.handleKeyCommand(editorState, command);
+    if (newState) {
+      this.onChange(newState);
+      return true;
+    }
+    return false;
+  }
+
+  _onTab(e) {
+    const maxDepth = 4;
+    this.onChange(RichUtils.onTab(e, this.state.editorState, maxDepth));
+  }
+
+  _toggleBlockType(blockType) {
+    this.onChange(
+      RichUtils.toggleBlockType(
+        this.state.editorState,
+        blockType
+      )
+    );
+  }
+
+  _toggleInlineStyle(inlineStyle) {
+    this.onChange(
+      RichUtils.toggleInlineStyle(
+        this.state.editorState,
+        inlineStyle
+      )
+    );
+  }
+
+  render() {
+    const {editorState} = this.state;
```

Comment:
Have we suddetly switched the whole project from 2 space indentation to 4 spaces? We need to keep it consistent.